### ISSUE TITLE
Simulation of boundary conditions and initial condition discontinuities

### ISSUE THREAD

**Issue by 979736316** [lululxvi/deepxde#625]
Dear Dr.Lu:
   Recently, I'm having a problem with PINNs simulation and would like to ask for your advice.  The problem is described as follows:
  **PDE:u_xx+u_x=u_t**
  **BC1:u(0,t)=0**
  **BC2:u(10,t)=m (where m is a constant)**
  **IC:   u(x,0)=np.piecewise(x, [x != 0, x == 0], [0, m]) (This means that the initial condition will be discontinuous at the point (10,0))**
   I simulated using PINNs and compared it with the analytical solution, and the results showed that all the loss terms were in the order of 1e-6 accuracy, and the final results are shown in the figure below. It is obvious that the error around the point (10, 0) is significantly higher than the other calculation areas, and the other points can be fitted better. Do you have any method to improve the accuracy of the calculation?
![微信图片_20220418161520](https://user-images.githubusercontent.com/52651366/163779129-dd6d857f-6cf8-402d-8c1d-d86b347d73ad.png)

**Comment by lululxvi** [comment]
If IC and BC are not consistent at the corner, then it is difficult, as NN is always a continuous function. You can modify IC and BC to be consistent.

**Comment by praksharma** [comment]
Here I solved a steady-state problem with 4 discontinuous corners with my own code (not DeepXDE).

![image](https://user-images.githubusercontent.com/70601302/164559784-0d7bbd17-5ba2-454b-9b0f-f49fda143cff.png)

You can't solve these problems with baseline PINN. A quick fix would be to not train those points. On top of that use importance sampling to increase the points cloud near the area.

Use other PINNs with skip connections that introduce artificial diffusion very similar to Petrov-Galerkin.

**Comment by 979736316** [comment]
> Here I solved a steady-state problem with 4 discontinuous corners with my own code (not DeepXDE). DeepXDE gains loss around the edges for some reason.
> 
> ![image](https://user-images.githubusercontent.com/70601302/164559784-0d7bbd17-5ba2-454b-9b0f-f49fda143cff.png)
> 
> You can't solve these problems with baseline PINN. A quick fix would be to not train those points. On top of that use importance sampling to increase the points cloud near the area.
> 
> Use other PINNs with skip connections that introduce artificial diffusion very similar to Petrov-Galerkin.

Thank you very much for your answer. For this, I tried RAR and adding calculation points in areas with large errors, and these results did not improve my calculation results. For the importance sampling you mentioned, do you mean adding training points in areas with large errors? Do you have similar learning documents, or can you show me the code with your correct results, I am curious how you handle it?

**Comment by praksharma** [comment]
Sorry, I can't share the code (we are writing a paper). You can learn more about importance sampling [here](https://arxiv.org/abs/2104.12325).

Their code is public but not open source, so we wrote our own version from scratch. Importance sampling is more than just adding training points in areas with large errors.

We are solving a similar 3D problem with discontinuities over the edges of a cuboid instead of just on the corners in 2D. Sorry, I wasn't helpful.

**Comment by 979736316** [comment]
> 
Thank you very much for your help and I look forward to your paper and available codes.

**Comment by 979736316** [comment]
> Sorry, I can't share the code (we are writing a paper). You can learn more about importance sampling [here](https://arxiv.org/abs/2104.12325).
> 
> Their code is public but not open source, so we wrote our own version from scratch. Importance sampling is more than just adding training points in areas with large errors.
> 
> We are solving a similar 3D problem with discontinuities over the edges of a cuboid instead of just on the corners in 2D. Sorry, I wasn't helpful.

Hi,  Is there an article on PINNs with skip connections？

**Comment by praksharma** [comment]
Wang's Learning rate annealing comprises a two transformer layer residual connection.

**Comment by 979736316** [comment]
Thank you for your answer
